packages/inngest/src/components/InngestCommHandler.ts

```diff
@@ -1199,14 +1136,75 @@ export class InngestCommHandler<
       }
 
       if (method === "PUT") {
-        let deployId = await actions.queryStringWithDefaults(
-          "processing deployment request",
-          queryKeys.DeployId
-        );
-        if (deployId === "undefined") {
-          deployId = undefined;
+        const [deployId, inBandSyncRequested] = await Promise.all([
+          actions
+            .queryStringWithDefaults(
+              "processing deployment request",
+              queryKeys.DeployId
+            )
+            .then((deployId) => {
+              return deployId === "undefined" ? undefined : deployId;
+            }),
+
+          Promise.resolve(
+            parseAsBoolean(this.env[envKeys.InngestAllowInBandSync])
+          )
+            .then((allowInBandSync) => {
+              if (!allowInBandSync) {
+                return syncKind.OutOfBand;
+              }
+
+              return actions.headers(
+                "processing deployment request",
+                headerKeys.InngestSyncKind
+              );
+            })
+            .then((kind) => {
+              return kind === syncKind.InBand;
+            }),
+        ]);
+
+        if (inBandSyncRequested) {
+          // Validation can be successful if we're in dev mode and did not
+          // actually validate a key. In this case, also check that we did indeed
+          // use a particular key to validate.
+          const sigCheck = await signatureValidation;
+
+          if (!sigCheck.success) {
+            return {
+              status: 401,
+              body: stringify({
+                code: "sig_verification_failed",
+                message: "",
+                name: "SigVerificationFailedError",
```

Comment:
@amh4r I used the same JSON as the Python SDK; not sure if this code is used anywhere else yet.

Reply:
I don't think it's used anywhere else. We should be consistent with the error code across SDKs but the name and message can be excluded